src/V3EmitCSyms.cpp

```diff
@@ -315,7 +318,8 @@ class EmitCSyms final : EmitCBaseVisitor {
             const string name_pretty = AstNode::vpiName(name);
             const int timeunit = m_modp->timeunit().powerOfTen();
             m_vpiScopeCandidates.insert(std::make_pair(
-                name, ScopeData(scopeSymString(name), name_pretty, timeunit, type)));
+                name, ScopeData(scopeSymString(name), name_pretty, timeunit, type,
+                                type == "SCOPE_MODULE" ? nodep->origModName() : "")));
```

Comment:
Is there a reason to not always pass origModName (make it unconditional)?  What sort of extra data leaks?

Reply:
I think this one leaks __BEGIN__

Reply:
I think exposing begin names would be compliant with IEEE, so I'd suggest we make it unconditional.